test/modules/paymentProcessor/PP_Connext_Crosschain_v1.t.sol

```diff
@@ -0,0 +1,1131 @@
+        mockWeth = address(weth);
+
+        // Deploy payment processor via clone
+        address impl = address(new PP_Connext_Crosschain_v1_Exposed());
+        paymentProcessor = PP_Connext_Crosschain_v1_Exposed(Clones.clone(impl));
+
+        _setUpOrchestrator(paymentProcessor);
+        _authorizer.setIsAuthorized(address(this), true);
+
+        // Initialize payment processor with config
+        bytes memory configData = abi.encode(mockEverClearSpoke, mockWeth);
+        paymentProcessor.init(_orchestrator, _METADATA, configData);
+
+        // Deploy and add payment client through timelock process
+        impl = address(new ERC20PaymentClientBaseV2Mock());
+        paymentClient = ERC20PaymentClientBaseV2Mock(Clones.clone(impl));
+        _orchestrator.initiateAddModuleWithTimelock(address(paymentClient));
+        vm.warp(block.timestamp + _orchestrator.MODULE_UPDATE_TIMELOCK());
+        _orchestrator.executeAddModule(address(paymentClient));
+
+        // Configure payment client
+        paymentClient.init(_orchestrator, _METADATA, bytes(""));
+        paymentClient.setIsAuthorized(address(paymentProcessor), true);
+        paymentClient.setToken(_token);
+
+        ORIGIN_CHAIN_ID = block.chainid;
+        TARGET_CHAIN_ID = 1337;
+
+        emptyExecutionData = new bytes32[](6);
+    }
+
+    //--------------------------------------------------------------------------
+    // Initialization Tests
+
+    /* Test initialization
+    */
+    function testInit() public override(ModuleTest) {
+        assertEq(
+            address(paymentProcessor.orchestrator()), address(_orchestrator)
+        );
+    }
+
+    /* Test interface support
+    */
+    function testSupportsInterface() public {
+        // Test for IModule_v1 interface
+        assertTrue(
+            paymentProcessor.supportsInterface(type(IModule_v1).interfaceId)
+        );
+
+        // Test for ICrossChainBase_v1 interface
+        assertTrue(
+            paymentProcessor.supportsInterface(
+                type(ICrossChainBase_v1).interfaceId
+            )
+        );
+
+        // Test for a non-supported interface (using a random interface ID)
+        assertFalse(paymentProcessor.supportsInterface(0xffffffff));
+    }
+
+    /* Test reinitialization
+    */
+    function testReinitFails() public override(ModuleTest) {
+        vm.expectRevert(OZErrors.Initializable__InvalidInitialization);
+        paymentProcessor.init(_orchestrator, _METADATA, abi.encode(1));
+    }
+
+    //--------------------------------------------------------------------------
+    // Payment Processing Tests
+
+    /* Test single payment processing
+    └── Given single valid payment order
+        └── When processing cross-chain payments
+            └── Then it should emit PaymentProcessed events for payment
+                └── And it should create cross-chain intent
+    */
+    function testFuzz_PublicProcessPayments_succeedsGivenSingleValidPaymentOrder(
```

Comment:
Naming convention for tests should be as following:

For public functions that succeed
`testFunctionName_worksGivenCondition()`

For public functions that should revert
`testFunctionName_revertGivenCondition()`

For internal functions that succeed
`testInternalFunctionName_worksGiveCondition()`

For internal function that should revert
`testInternalFunctionName_revertGivenCondition()`

You can reference the tests for the FM or LM on the other work done by audit33 to see how the functions should be formatted, same for the Gherkin

Reply:
Awesome thanks for this I just pushed up some changes would you mind checking them when you get a chance? I think I was able to follow the naming convention properly but want to be sure we're aligned let me know me know if there's any feedback and I'll make the changes myself thanks!